icu4c/source/common/rbbitblb.cpp

```diff
@@ -1378,18 +1392,29 @@ void RBBITableBuilder::exportTable(void *where) {
     for (state=0; state<table->fNumStates; state++) {
         RBBIStateDescriptor *sd = (RBBIStateDescriptor *)fDStates->elementAt(state);
         RBBIStateTableRow   *row = (RBBIStateTableRow *)(table->fTableData + state*table->fRowLen);
-        U_ASSERT (-32768 < sd->fAccepting && sd->fAccepting <= 32767);
-        U_ASSERT (-32768 < sd->fLookAhead && sd->fLookAhead <= 32767);
-        row->fAccepting = (int16_t)sd->fAccepting;
-        row->fLookAhead = (int16_t)sd->fLookAhead;
-        row->fTagIdx    = (int16_t)sd->fTagsIdx;
-        for (col=0; col<catCount; col++) {
-            row->fNextState[col] = (uint16_t)sd->fDtran->elementAti(col);
+        if (use8BitsForTable()) {
+            U_ASSERT (-128 < sd->fAccepting && sd->fAccepting < kStateThresholdFor8BitsTable);
+            U_ASSERT (-128 < sd->fLookAhead && sd->fLookAhead < kStateThresholdFor8BitsTable);
+            U_ASSERT (-128 < sd->fTagsIdx && sd->fTagsIdx <= kStateThresholdFor8BitsTable);
+            row->s8.fAccepting = (int8_t)sd->fAccepting;
+            row->s8.fLookAhead = (int8_t)sd->fLookAhead;
+            row->s8.fTagIdx    = (int8_t)sd->fTagsIdx;
+            for (col=0; col<catCount; col++) {
+                row->s8.fNextState[col] = (int8_t)sd->fDtran->elementAti(col);
+            }
+        } else {
+            U_ASSERT (-32768 < sd->fAccepting && sd->fAccepting <= 32767);
+            U_ASSERT (-32768 < sd->fLookAhead && sd->fLookAhead <= 32767);
+            row->s16.fAccepting = (int16_t)sd->fAccepting;
+            row->s16.fLookAhead = (int16_t)sd->fLookAhead;
+            row->s16.fTagIdx    = (int16_t)sd->fTagsIdx;
```

Comment:
FYI why is it called fTagIdx (singular tag) in one struct and fTagsIdx (plural tags) in the other?

Reply:
noted for follow-up